src/control_position.c

```diff
@@ -79,16 +86,24 @@ void control_position_update(control_position_t* cpos, uint32_t now_us) {
 		-shifted_vel_coast, shifted_vel_coast
 	);
 
+	float new_vel_output_unshifted = 0;
 	if (fabs(cpos->prop_out) > fabs(cpos->vel_output_unshifted) || cpos->prop_out * cpos->vel_output_unshifted < 0) {
-		cpos->vel_output_unshifted = linear_ramp_to(
+		new_vel_output_unshifted = linear_ramp_to(
 			cpos->vel_output_unshifted,
 			cpos->cfg->acceleration * dt,
 			cpos->prop_out
 		);
 	} else {
-		cpos->vel_output_unshifted = cpos->prop_out;
+		new_vel_output_unshifted = cpos->prop_out;
 	}
 
+	if (cpos->direction_changed && signf(new_vel_output_unshifted) != signf(cpos->vel_output_unshifted)) {
+		cpos->changed_direction_wait_timestamp = cpos->now_us;
+		cpos->direction_changed = false;
```

Comment:
Actually, why do we need to set `direction_changed` in `control_position_target_pos`? I would propose trying the following instead:
- Set it to `true` here, instead of `false`, so that it is true while we're waiting for the timer to expire
- In the start of update, do the timer check only if `direction_changed` is `true`, and then set it to `false` when the timer expires. This should protect against timer overflows, which happen exactly every 1hr 11min 34.97sec
- Remove the added code from `control_position_target_pos`

If there is a reason for this not to work, which I'm not seeing, maybe we need to add an additional flag to guard against timer overflows...